### PR TITLE
Add on-ride photo sections to APVC and Reverse Freefall Coaster

### PR DESCRIPTION
**Pull request by Fred-1044276**
This adds on-ride photo sections to the Air Powered Vertical Coaster and Reverse Freefall Coaster. They are enabled by default and will cost a fixed $300 to build in-game. They will make these coasters even better, esp. in "Pay to enter park / Free rides" scenarios where they normally can't even make any money at all.

I might have messed up somewhere when editing the drawing code to add the on-ride photo sections, but edits to fix the on-ride photo section code for these coasters are welcomed.

### CODE REVIEW

File: src/openrct2/ride/coaster/AirPoweredVerticalCoaster.cpp [hunk 1]
```diff
@@ -929,6 +929,27 @@ static void air_powered_vertical_rc_track_booster(
     paint_util_set_general_support_height(session, height + 32, 0x20);
 }
 
+static void air_powered_vertical_rc_track_on_ride_photo(
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/AirPoweredVerticalCoaster.cpp:932]: What did you use as a base to implement this function?

File: src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp [hunk 2]
```diff
@@ -385,6 +385,29 @@ static void paint_reverse_freefall_rc_vertical(
     }
 }
 
+static void paint_reverse_freefall_rc_on_ride_photo(
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp:388]: You're drawing an on-ride photo section with LIM boosters here, even though it doesn't actually accelerate the train. You will need to draw it with a non-booster track (use the station track for this).

File: src/openrct2/ride/coaster/AirPoweredVerticalCoaster.cpp (hunk 1 above)
Comment (Fred-1044276) [reply src/openrct2/ride/coaster/AirPoweredVerticalCoaster.cpp:932]: I used the APVC's straight track pieces and the splash boats' on-ride photo as a base.

File: src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp (hunk 2 above)
Comment (Fred-1044276) [reply src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp:388]: RFC
Would this work? This is derived from the station track piece drawing code, except I have removed all references to the station platform drawing and station identification codes.

File: src/openrct2/ride/coaster/AirPoweredVerticalCoaster.cpp [hunk 3]
```diff
@@ -929,6 +929,27 @@ static void air_powered_vertical_rc_track_booster(
...
+    paint_session* session, ride_id_t rideIndex, uint8_t trackSequence, uint8_t direction, int32_t height,
+    const TileElement* tileElement)
+{
+    static constexpr const uint32_t imageIds[4] = {
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/AirPoweredVerticalCoaster.cpp:936]:
```suggestion
    static constexpr const std::array<uint32_t, 4> imageIds = {
```

File: src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp [hunk 4]
```diff
@@ -385,6 +385,27 @@ static void paint_reverse_freefall_rc_vertical(
     }
 }
 
+static void reverse_freefall_rc_on_ride_photo(
+    paint_session* session, ride_id_t rideIndex, uint8_t trackSequence, uint8_t direction, int32_t height,
+    const TileElement* tileElement)
+{
+    static constexpr const uint32_t imageIds[4] = {
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp:392]:
```suggestion
    static constexpr const std::array<uint32_t, 4> imageIds = {
```

File: src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp]: (This has since been discussed on Discord.)

**Comment by Gymnasiast** [comment]
Please fix compilation and (especially!) test your work. We want PRs to be tested by their authors first, because reviewing is hard enough.

File: src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2/ride/coaster/ReverseFreefallCoaster.cpp:392]: Also, put a comment above explaining why we're using the station pieces here.

**Comment by Fred-1044276** [comment]
Test Park 2020-05-10 17-18-21
I did some testing by building the game with the updated files from this PR, loading up a scenario with the APVC and RFC available, and actually constructing them with the on-ride photo sections. It works.

**Comment by Gymnasiast** [comment]
> I did some testing by building the game with the updated files from this PR, loading up a scenario with the APVC and RFC available, and actually constructing them with the on-ride photo sections. It works.

Did you copy and paste the changes or did you actually check out the branch? It looks like you did the former, since it fails compilation on everything.

You are quick to chase us to review your pull requests, but you should only do so if your stuff actually compiles. Otherwise it's just a wasted effort on our part to even *look* at your PR. And I get the idea that you don't really pay that much attention to any instructions we give.

**Comment by Fred-1044276** [comment]
What now? Should I just start over? I still have the edited files on my computer

**Comment by Gymnasiast** [comment]
You need to make sure this branch builds. Normally you would check out the branch on your computer, make the changes necessary (on your own system, not via Github), build it, commit these changes, then push that again. In other words: the editing workflow should be on your own system.

**Comment by Gymnasiast** [comment]
Since there has been no reply in 3 weeks and this PR does not compile, I have decided to close it. I created https://github.com/OpenRCT2/OpenRCT2/issues/11859 , so someone else can have a go at it.

**Pull request closed without merging**